components/match2/wikis/hearthstone/match_summary.lua

```diff
@@ -32,39 +37,150 @@ end
+		local isWinner = opponentIndex == game.winner or game.resultType == 'draw'
+		if game.resultType == 'default' then
+			return OpponentDisplay.BlockScore{
+				isWinner = isWinner,
+				scoreText = isWinner and 'W' or string.upper(game.walkover),
+			}
+		end
+
+		local score = game.resultType ~= 'np' and (game.scores or {})[opponentIndex] or nil
+		return OpponentDisplay.BlockScore{
+			isWinner = isWinner,
+			scoreText = score,
+		}
+	end
+
+	return HtmlWidgets.Div{
+		classes = {'brkts-popup-header-dev'},
+		css = {['justify-content'] = 'center', margin = 'auto'},
+		children = WidgetUtil.collect(
+			HtmlWidgets.Div{
+				classes = {'brkts-popup-header-opponent', 'brkts-popup-header-opponent-left'},
+				children = {
+					createOpponent(1),
+					createScore(1):addClass('brkts-popup-header-opponent-score-left'),
+				},
+			},
+			HtmlWidgets.Div{
+				classes = {'brkts-popup-header-opponent', 'brkts-popup-header-opponent-right'},
+				children = {
+					createScore(2):addClass('brkts-popup-header-opponent-score-right'),
+					createOpponent(2),
+				},
+			}
+		)
+	}
+end
 
+---@param isTeamMatch boolean
+---@param game MatchGroupUtilGame
+---@param gameIndex number
+---@return Widget
+function CustomMatchSummary._createGame(isTeamMatch, game, gameIndex)
 	return MatchSummaryWidgets.Row{
 		classes = {'brkts-popup-body-game'},
-		css = {['font-size'] = '80%', padding = '4px', ['min-height'] = '24px'},
+		css = {padding = '4px', ['min-height'] = '24px'},
 		children = WidgetUtil.collect(
-			MatchSummaryWidgets.Characters{characters = team1Characters, flipped = false},
+			CustomMatchSummary._displayOpponents(isTeamMatch, game.opponents[1].players, true),
 			MatchSummaryWidgets.GameWinLossIndicator{winner = game.winner, opponentIndex = 1},
+			MatchSummaryWidgets.GameCenter{css = {['font-size'] = '80%'}, children = 'Game ' .. gameIndex},
 			MatchSummaryWidgets.GameWinLossIndicator{winner = game.winner, opponentIndex = 2},
-			MatchSummaryWidgets.Characters{characters = team2Characters, flipped = true}
+			CustomMatchSummary._displayOpponents(isTeamMatch, game.opponents[2].players)
 		)
 	}
 end
 
+---@param isTeamMatch boolean
+---@param players table[]
+---@param flip boolean?
+---@return Html?
+function CustomMatchSummary._displayOpponents(isTeamMatch, players, flip)
+	local playerDisplays = Array.map(players, function (player)
+		local class = player.class
+		if Logic.isEmpty(class) then
+			return
+		end
+		local playerWrapper = mw.html.create('div')
+			:css('display', 'flex')
+			:css('flex-direction', flip and 'row-reverse' or 'row')
+			:css('gap', '2px')
+			:css('width', '100%')
+
+		playerWrapper:node(HtmlWidgets.Div{
+			classes = {'brkts-champion-icon'},
+			children = MatchSummaryWidgets.Character{
+				character = class,
+				showName = not isTeamMatch,
+				flipped = flip,
+			}
+		})
+
+		if isTeamMatch then
+			playerWrapper:node(PlayerDisplay.BlockPlayer{player = player, flip = flip})
+		end
+
+		return playerWrapper
```

Comment:
```suggestion
		local char = HtmlWidgets.Div{
			classes = {'brkts-champion-icon'},
			children = MatchSummaryWidgets.Character{
				character = class,
				showName = not isTeamMatch,
				flipped = flip,
			}
		}

		return HtmlWidgets.Div{
			css = {
				display = 'flex',
				['flex-direction'] = flip and 'row-reverse' or 'row',
				gap = '2px',
				width = '100%'
			},
			children = {
				char,
				isTeamMatch and PlayerDisplay.BlockPlayer{player = player, flip = flip} or nil,
			},
		}
```
untested, but would port the rest to widget ;)

Reply:
works fine :D